client/src/plugins/error-tracking/ErrorTracking.js

```diff
@@ -0,0 +1,198 @@
+/**
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.
+ *
+ * Camunda licenses this file to you under the MIT; you may not use this file
+ * except in compliance with the MIT License.
+ */
+
+import debug from 'debug';
+
+import { PureComponent } from 'react';
+
+import * as Sentry from '@sentry/browser';
+
+import Metadata from '../../util/Metadata';
+
+import Flags, { SENTRY_DSN, DISABLE_REMOTE_INTERACTION } from '../../util/Flags';
+
+const PRIVACY_PREFERENCES_CONFIG_KEY = 'editor.privacyPreferences';
+const EDITOR_ID_CONFIG_KEY = 'editor.id';
+const CRASH_REPORTS_CONFIG_KEY = 'ENABLE_CRASH_REPORTS';
+const NON_EXISTENT_EDITOR_ID = 'NON_EXISTENT_EDITOR_ID';
+
+const log = debug('ErrorTracking');
+
+const ONE_MINUTE_MS = 1000 * 60;
+
+// DSN is set to Travis as an env variable, passed to client via WebPack DefinePlugin
+const DEFINED_SENTRY_DSN = process.env.SENTRY_DSN;
+
+export default class ErrorTracking extends PureComponent {
+
+  constructor(props) {
+    super(props);
+
+    // SENTRY_DSN flag is useful for development.
+    this.SENTRY_DSN = Flags.get(SENTRY_DSN) || DEFINED_SENTRY_DSN;
+
+    // Setting this here so that we can mock later if necessary.
+    this.SCHEDULE_TIME = ONE_MINUTE_MS;
+
+    // Setting this here so that we can mock later if necessary.
+    this._sentry = Sentry;
+
+    this._isInitialized = false;
+  }
+
+  async componentDidMount() {
+    const { result, msg } = await this.canInitializeSentry();
+
+    if (!Flags.get(DISABLE_REMOTE_INTERACTION)) {
+
+      // If remove interaction is not disabled via flags:
+      // -> The user may turn on / off error reporting on the run
+      // -> The user may never actually restart the modeler.
+      // That's why we'll schedule a check and turn on / off Sentry if necessary.
+      this.scheduleCheck();
+    }
```

Comment:
What caught my attention here is that you need to know that `this.canInitializeSentry()` returns `result=false` when the flag is not set. Otherwise one could suspect that we schedule check even when we initialize sentry later.

So my suggestion is to do an early return if the flag is not set.

Reply:
So I'm not sure if I understood your point correctly but we **should** reschedule a check even though we initialize Sentry because the user may turn the "Error Tracking" option off via "Privacy Preferences" later.

The only exception is that: DISABlE_REMOTE_INTERACTION flag is set -> We can never adjust a flag value on the run. If the user does not want remove interaction, we should not reshcedule a check as the result of `canInitializeSentry` would never change on the run.

Reply:
What I mean is:

```
    const { result, msg } = await this.canInitializeSentry();

    if (!Flags.get(DISABLE_REMOTE_INTERACTION)) {

      // If remove interaction is not disabled via flags:
      // -> The user may turn on / off error reporting on the run
      // -> The user may never actually restart the modeler.
      // That's why we'll schedule a check and turn on / off Sentry if necessary.
      this.scheduleCheck();
      return;
    }
```

Reply:
In this case we won't be able to log why we're not initializing Sentry.

Few lines below we do an early return after logging the reason why we cannot initialize Sentry.

Reply:
I think the code is fine like this. It does the job and it's simple enough.

Reply:
I mean I actually had to double-check if we skip initialization when we schedule the check. How about if we just schedule the check in the `if (!result)` block in case the flag condition is met?

Reply:
**We don't want to**  schedule a check only if `canInitializeSentry` returns false (AKA. `if(!result) block`)

**We want to** schedule a check as long as remote interaction is not disabled via flag. Scheduling does not depend on the outcome of `canInitializeSentry`.

We cannot initialize Sentry for now? Fine -> Let's schedule a check and see if we initialize it later (the user turned on Error Tracking option)

We can initialize Sentry for now? Fine -> Let's schedule a check and see if we need to close it later (the user turned off  Error Tracking option)

Reply:
So it's totally unrelated to the result of `canInitializeSentry`, right? Then I would move it so that it's not in the middle of the flow where we check if we can initialize Sentry.

Reply:
Refactored via 81ecccaf6b2efaea00b537f76d0e1e793dc274a9